ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/MaximumDrawdownCriterion.java

```diff
@@ -35,25 +35,25 @@ this software and associated documentation files (the "Software"), to deal in
 public class MaximumDrawdownCriterion extends AbstractAnalysisCriterion {
 
     @Override
-    public double calculate(TimeSeries series, TradingRecord tradingRecord) {
+    public Num calculate(TimeSeries series, TradingRecord tradingRecord) {
         CashFlow cashFlow = new CashFlow(series, tradingRecord);
         Num maximumDrawdown = calculateMaximumDrawdown(series, cashFlow);
-        return maximumDrawdown.doubleValue();
+        return maximumDrawdown;
     }
 
     @Override
-    public double calculate(TimeSeries series, Trade trade) {
+    public Num calculate(TimeSeries series, Trade trade) {
         if (trade != null && trade.getEntry() != null && trade.getExit() != null) {
             CashFlow cashFlow = new CashFlow(series, trade);
             Num maximumDrawdown = calculateMaximumDrawdown(series, cashFlow);
```

Comment:
MaximumDrawdown looks good.  Interesting that `calculateMaximumDrawdown()` was already converted to `Num`..